### ISSUE TITLE
Error when creating projects using system robot account

### ISSUE THREAD

**Issue by coding4food** [goharbor/terraform-provider-harbor#418]
**Describe the bug**
When using API with system robot account credentials terraform creates project but returns error:
```
harbor_project.harbor_project: Creating...
╷
│ Error: [ERROR] unexpected status code got: 403 expected: 200 
│  {"errors":[{"code":"FORBIDDEN","message":"forbidden"}]}
│ 
│ 
│   with harbor_project.harbor_project,
│   on main.tf line 67, in resource "harbor_project" "harbor_project":
│   67: resource "harbor_project" "harbor_project" {
```

**To Reproduce**
1. Use the following terraform config:
```terraform
terraform {
  required_providers {
    harbor = {
      source  = "goharbor/harbor"
      version = "3.10.8"
    }
  }
}

provider "harbor" {
  url = "https://reg.example.com"
}

resource "harbor_robot_account" "registry_robot_system" {
  name        = "test-robot-system"
  description = "test"
  level       = "system"
  duration    = 30

  permissions {
    kind      = "system"
    namespace = "/"

    access {
      resource = "project"
      action = "*"
    }
  }
}

provider "harbor" {
  alias = "robot-system"
  url = "https://reg.example.com"
  username = harbor_robot_account.registry_robot_system.full_name
  password = harbor_robot_account.registry_robot_system.secret
}

resource "harbor_project" "harbor_project" {
  name                   = "test-app-fff-123"
  public                 = false
  vulnerability_scanning = false

  provider = harbor.robot-system
}
```
2. Run `terraform apply`

**Expected behavior**
Terraform creates Harbor project without errors.

**Additional context**
Relates to https://github.com/goharbor/harbor/issues/18363

Harbor instance is configured so that only sysadmins can create projects.

- Provider Version: 3.10.0/3.10.8
- Terraform Version: 1.5.2
- Harbor Version: 2.9.0

**Comment by flbla** [comment]
Hi,
I think your robot account is missing some rights on the kind "project".
I created a robot account with below RBAC and it seems to work : 
```
    "permissions": [
      {
        "access": [
          {
            "action": "delete",
            "resource": "project"
          },
          {
            "action": "read",
            "resource": "project"
          },
          {
            "action": "update",
            "resource": "project"
          }
        ],
        "kind": "project",
        "namespace": "*"
      },
      {
        "access": [
          {
            "action": "create",
            "resource": "project"
          },
          {
            "action": "list",
            "resource": "project"
          }
        ],
        "kind": "system",
        "namespace": "/"
      }
    ],
```

on the kind system only create and list are possible :
![image](https://github.com/goharbor/terraform-provider-harbor/assets/11298399/d0026226-e9fd-41b6-b1d1-1a080ac0aac5)


let me know if it works for you.

**Comment by flbla** [comment]
Hi,
did you try my RBAC ? because I tested it and it works, I was able to create/read/destroy project with it

the data source uses a different endpoint because it needs to find the project from its name
as the resource already have the ID of the project it doesn't need to look for it.
I would prefer to keep it like this, I don't want to handle such workaround in the provider

@YangJiao0817 / @wy65701436 : any advice on the Harbor RBAC ? thanks

**Comment by coding4food** [comment]
@flbla, it appears that problem has been fixed in Harbor 2.10. Thanks!
